core.py

```diff
@@ -80,16 +106,18 @@ def __init__(self, tof_lib):
         else:
             self.arduino = None
             self.PWMservo = PWM.Servo(pulse_incr_us=1)
-            for pin in range(0, 3):
-                i2c_lidar.xshut([LIDAR_PINS[pin]])
-                self.lidars.append(
-                    i2c_lidar.create(LIDAR_PINS[pin], tof_lib, 0x2a + pin)
-                )
+
+        # I2C lidar sensors should be enabled
+        # in both Arduino mode or direct pi mode.
+        for pin in range(0, 3):
+            i2c_lidar.xshut([LIDAR_PINS[pin]])
```

Comment:
also as per [above comment ](https://github.com/hackhitchin/piwars3/pull/17#discussion_r107375706),there's no need for the additional `for..in range` construct. Just jump straight in with a `[i2c_lidar.xshut(LIDAR_PINS[pin]) for pin in LIDAR_PINS]` to iterate the list assigning pins.